### PR TITLE
Fix item reactions not being set for EXPERIMENTOR

### LINKED ISSUE #71553: Saving on experimentor initialize time made the experimentor never generate valid experiment types for items (except relics)
## What
In #71007, `SetTypeReactions()` proc that was remade into a `valid_items()` proc was removed from initialization. It was be set to be called only on a successful medium chance irradiate transformative reaction. The problem is that the proc also runs assignment of valid reactions for all items. `item_reactions` var would be never set.
Thus no items have a reaction type that would have a successful match.
And since no items have a valid reaction, there is no item that has a valid irradiate reaction, so `valid_items()` will never get called.
Now you will always either roll a fail state of two random sounds, or a global malfunction reaction. Relics are an exception because they are hardcoded to always display and do the discover reaction when inserted.
The only way to get a reaction on an item now is to roll a global malfunction that forcepicks a reaction for the inserted item type.

https://github.com/tgstation/tgstation/blob/1900eb9484ddfd57ed37c9b7b21bee77b96b8205/code/modules/research/experimentor.dm#L82-L95

https://github.com/tgstation/tgstation/blob/1900eb9484ddfd57ed37c9b7b21bee77b96b8205/code/modules/research/experimentor.dm#L297-L307

https://github.com/tgstation/tgstation/blob/1900eb9484ddfd57ed37c9b7b21bee77b96b8205/code/modules/research/experimentor.dm#L50-L80

### PR DESCRIPTION
**Pull request by Mothblocks**
Fixes #71553

## Changelog
:cl:
fix: Fixed reactions being broken in the EXPERIMENTOR or something I don't actually know what this thing does
/:cl:

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
Would it be slightly more performant to generate the item reactions as needed instead of all at once for everything? You could probably get away with setting up a placeholder blacklist reaction type for anything you find to be on the blacklist and just null checking for whether the reaction type has already been generated.

**Comment by Mothblocks** [comment]
The performance difference there isn't super relevant, given this is an already very cold path, but also one that can only be run once. Also this is a 37ms proc, it was optimized alongside a dozen other procs in that range, but in the context of a one-and-done call in game, it's not bad.

File: code/modules/research/experimentor.dm [hunk 1]
```diff
@@ -48,14 +48,28 @@
...
+
+/obj/machinery/rnd/experimentor/proc/item_reactions()
+	RETURN_TYPE(/list)
+
+	if (isnull(item_reactions))
+		generate_valid_items_and_item_reactions()
+
+	return item_reactions
+
+/obj/machinery/rnd/experimentor/proc/generate_valid_items_and_item_reactions()
```
Comment (Time-Green) [inline code/modules/research/experimentor.dm:66]: Could you autodoc the two new procs?

Comment (Mothblocks) [reply code/modules/research/experimentor.dm:66]: Sure but for note auto-docs are not necessary unless implementors use them

Comment (Time-Green) [reply code/modules/research/experimentor.dm:66]: You're technically right but this whole thing is a wasteland tbh

**Comment by Time-Green** [comment]
It's not required so blocking it is kinda pointless and mean tbh

**Pull request merged**